Emby.Server.Implementations/EntryPoints/ExternalPortForwarding.cs

```diff
@@ -15,179 +14,104 @@
 
 namespace Emby.Server.Implementations.EntryPoints
 {
+    /// <summary>
+    /// Server entrypoint handling external port forwarding.
+    /// </summary>
     public class ExternalPortForwarding : IServerEntryPoint
     {
         private readonly IServerApplicationHost _appHost;
         private readonly ILogger _logger;
-        private readonly IHttpClient _httpClient;
         private readonly IServerConfigurationManager _config;
         private readonly IDeviceDiscovery _deviceDiscovery;
 
+        private readonly object _createdRulesLock = new object();
+        private List<IPEndPoint> _createdRules = new List<IPEndPoint>();
         private Timer _timer;
+        private string _lastConfigIdentifier;
 
-        private NatManager _natManager;
-
-        private readonly object _createdRulesLock = new object();
-        private List<string> _createdRules = new List<string>();
-        private readonly object _usnsHandledLock = new object();
-        private List<string> _usnsHandled = new List<string>();
+        private bool _disposed = false;
 
-        public ExternalPortForwarding(ILoggerFactory loggerFactory, IServerApplicationHost appHost, IServerConfigurationManager config, IDeviceDiscovery deviceDiscovery, IHttpClient httpClient)
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExternalPortForwarding"/> class.
+        /// </summary>
+        /// <param name="logger">The logger.</param>
+        /// <param name="appHost">The application host.</param>
+        /// <param name="config">The configuration manager.</param>
+        /// <param name="deviceDiscovery">The device discovery.</param>
+        public ExternalPortForwarding(
+            ILogger<ExternalPortForwarding> logger,
+            IServerApplicationHost appHost,
+            IServerConfigurationManager config,
+            IDeviceDiscovery deviceDiscovery)
         {
-            _logger = loggerFactory.CreateLogger("PortMapper");
+            _logger = logger;
             _appHost = appHost;
             _config = config;
             _deviceDiscovery = deviceDiscovery;
-            _httpClient = httpClient;
-            _config.ConfigurationUpdated += _config_ConfigurationUpdated1;
```

Comment:
You removed the listener here but kept it in the RunAsync method, and also allocate a new one every time in the method rather than replace the old one. I'm not too familiar with C# but was the old behavior incorrect?

Reply:
@Bond-009 wouldn't that still add new listeners every time that function gets called without removing the old ones? That depends on whether RunAsync can get called more than once before the Dispose method gets called.

Reply:
`RunAsync` only gets called once, op startup.